core/tests/test_utils.py

```diff
@@ -3922,26 +4516,32 @@ def __init__(self, f, exception, num_tries_before_success):
         self._exception = exception
         self._num_tries_before_success = num_tries_before_success
         self._always_fail = (
-            self._num_tries_before_success == FailingFunction.INFINITY)
+            str(self._num_tries_before_success) == FailingFunction.INFINITY)
         self._times_called = 0
 
-        if not self._always_fail and self._num_tries_before_success < 0:
+        if (
+            not self._always_fail and
+            isinstance(self._num_tries_before_success, int) and
+            self._num_tries_before_success < 0
+        ):
             raise ValueError(
                 'num_tries_before_success should either be an '
                 'integer greater than or equal to 0, '
                 'or FailingFunction.INFINITY')
 
-    def pre_call_hook(self, args):
+    # Here we use type Any because argument 'args' can accept arbitrary number
+    # of function's arguments and these arguments can be of any type.
+    def pre_call_hook(self, args: OrderedDict[str, Any]) -> None:
         """Method that is called each time before the actual function call to
         check if the exception is to be raised based on the number of tries
         before success.
 
         Args:
-            args: list(*). Set of arguments this function accepts.
+            args: OrderedDict. Set of arguments that the function accepts.
         """
         self._times_called += 1
         call_should_fail = (
-            self._always_fail or
-            self._num_tries_before_success >= self._times_called)
+            self._always_fail or isinstance(self._num_tries_before_success, int)
+            and self._num_tries_before_success >= self._times_called)
```

Comment:
Can you properly parentheses the `or` or `and`?

Reply:
Did some changes PTAL!